bazel/repositories.bzl

```diff
@@ -186,21 +177,15 @@ def envoy_dependencies(path = "@envoy_deps//", skip_protobuf_bzl = False, skip_t
         recipes = recipes.to_list(),
     )
 
-    protobuf_repository = repository_rule(
-        implementation = _protobuf_repository_impl,
-        attrs = {
-            "envoy_deps_path": attr.string(),
-        },
-    )
-
-    # If the WORKSPACE hasn't already supplied @protobuf_bzl and told us to skip it, we need to map in the
-    # full repo into @protobuf_bzl so that we can depend on this in envoy_build_system.bzl and for things
-    # like @protobuf_bzl//:cc_wkt_protos in envoy-api. We do this by some evil symlink stuff.
-    if not skip_protobuf_bzl:
-        protobuf_repository(
-            name = "protobuf_bzl",
-            envoy_deps_path = path,
-        )
+    existing_rule_keys = native.existing_rules().keys()
```

Comment:
Why is `existing_rule_keys` needed? Please add a comment if it's necessary

Reply:
It lets users override the dependencies simply by defining their own, instead of having to adjust the params of `envoy_dependencies`. For example, I could have a `WORKSPACE` that uses a custom `protobuf_bzl`, and then this code would use it automatically.

I'll add a comment.

Reply:
It seems crazy to have two mechanisms for this (this and the `skip_targets`). I don't feel that strongly on which is better, but we should just have one, even if we need to deprecate `skip_targets`.

Reply:
I would prefer `skip_targets`, since there some optional dependencies that can be skipped without overriding, e.g. lightstep.

Reply:
@htuch I'd like to deprecate `skip_targets` once the migration is finished, because it'll be redundant and is harder to use when Envoy is built as part of a larger workspace.

@lizan If a dependency is optional, then Bazel won't try to fetch it unless it's needed to build a target the user asked for.

Reply:
@lizan can we get rid of `skip_targets` based on this? Would really prefer just one approach here.

Reply:
ah you're right, I'm fine with that.